### PR TITLE
use `asyncio.wait`'s `timeout` kwarg for `task_max_timeout`

### PR DESCRIPTION
**Pull request by escapewindow**
In #242 , we noticed scriptworker 13.0.0 introduced a hang in running tasks. Because the timeout task now slept for the entire `task_max_timeout` and was one of the tasks we were awaiting, all tasks took the entire `task_max_timeout` time to complete.

This patch refactors `run_task` to use the builtin `timeout` kwarg in `asyncio.wait`. If there are pending futures, we `kill_proc`, which raises a `ScriptWorkerTaskException`. We then clean up in the `finally` block.

This allows us to remove the `max_timeout` and `get_future_exception` functions.

I don't want to roll this out and then go on a long PTO, so I'm currently thinking I'll leave this for review, and roll this out when I get back (assuming it passes review). If we need to release a new scriptworker in the meantime, we should branch off the [12.1.0 tag](https://github.com/mozilla-releng/scriptworker/releases/tag/12.1.0).

...aand it looks like I need a followup fix for the intermittent bustage. I think I'll fix by explicitly passing in an event loop into certain functions from the tests.

### CODE REVIEW

**Comment by escapewindow** [comment]
Alternate solutions were here:

* https://github.com/escapewindow/scriptworker/commits/first-completed
* https://github.com/escapewindow/scriptworker/commits/cancel-callback
* and the original `asyncio.call_later(timeout, max_timeout, *args)` in scriptworker 12.1.0 and earlier.

**Comment by escapewindow** [comment]
@Callek , @tomprince what do you think?
As mentioned in the [original comment](https://github.com/mozilla-releng/scriptworker/pull/243#issue-199581863), I think this deploy will probably wait until after I get back on July 16.

File: scriptworker/test/test_task.py [hunk 1]
```diff
@@ -297,14 +298,51 @@ def test_prepare_to_run_task(context):
...
+    # create all 6 files
+    context.config['task_max_timeout'] = 5
+
+    pre = arrow.utcnow().timestamp
+    with pytest.raises(ScriptWorkerTaskException):
+        await swtask.run_task(context)
+    post = arrow.utcnow().timestamp
+    # I don't love these checks, because timing issues may cause this test
+    # to be flaky. However, I don't want a non- or long- running test to pass.
+    # Did this run at all?
```
Comment (Callek) [inline scriptworker/test/test_task.py]: could python freezegun help here?

Comment (escapewindow) [reply scriptworker/test/test_task.py]: I may be misreading, but afaict, freezegun makes the datetime calls return a predictable time. How would we know if our timeout task took too long or short if both calls return the same timestamp?

Comment (Callek) [reply scriptworker/test/test_task.py]: I was imagining using it for https://github.com/spulec/freezegun#manual-ticks and/or https://github.com/spulec/freezegun#moving-time-to-specify-datetime

Comment (escapewindow) [reply scriptworker/test/test_task.py]: The time tests in this test are specifically to measure how long we're running the task for before killing: we need to know that the task runs for at least as long as max timeout, but not significantly longer (e.g., if the task runs to completion without being killed, that should be an error). Manually changing the timestamps to pass means those tests do nothing useful, and we should remove them. I know that timestamp tests are potentially flaky, but I also want to know if the kill isn't working properly.

Comment (Callek) [reply scriptworker/test/test_task.py]: Fair, I was musing that it may help, not blocking on it, to be extra clear

Comment (escapewindow) [reply scriptworker/test/test_task.py]: Ok. Thanks for the pointer; looks like that should be helpful in the future for other time-based tests.

I'm going to push scriptworker 14.0.0 changes here and then merge.

**Pull request merged**